docs/lang/articles/basic/field.md

```diff
@@ -14,42 +14,170 @@ A 0D (zero-dimensional) field contains *only* one element.
 
 
 ### Declaration
 
-```python
-import taichi as ti
-ti.init(arch=ti.cpu)
+The simplest way to declare a scalar field is to call `ti.field(dtype, shape)`, where `dtype` is a primitive data type as explained in [type system](../type_system/type.md) and `shape` is a tuple of integers.
+
++ To declare a 0D scalar field, set its shape to the empty tuple `()`.
+
+    ```python
+    # Declare a 0D scalar field whose data type is f32
+    f_0d = ti.field(ti.f32, shape=())  # 0D field
+    ```
+
+    The memory layout of the field `f_0d` will be like:
+
+    ```
+        ┌─────┐
+        │     │
+        └─────┴
+        └─────┘
+     f_0d.shape=()
+    ```
+
++ To declare a 1D scalar field of length `n`, set its shape to `n` or `(n,)`:
+
+    ```python
+    f_1d = ti.field(ti.i32, shape=9)  # 1D field
+    ```
+
+    The memory layout of the field `f_1d` will be like:
+
+    ```
+    ┌───┬───┬───┬───┬───┬───┬───┬───┬───┐
+    │   │   │   │   │   │   │   │   │   │
+    └───┴───┴───┴───┴───┴───┴───┴───┴───┘
+    └───────────────────────────────────┘
+            f_1d.shape = (9,)
+    ```
+
++ To declare a 2D scalar field, set the size of its first two dimensions, i.e. the number of rows and columns, respectively. For example, the following code defines a 2D scalar field of shape (3, 6), which has 3 rows and 6 columns:
+
+    ```python
+    f_2d = ti.field(int, shape=(3, 6))  # 2D field
+    ```
+
+    The memory layout of the field `f_2d` will be like:
+
+    ```
+                           f_2d.shape[1]
+                               (=6)
+                     ┌───────────────────────┐
+
+                  ┌  ┌───┬───┬───┬───┬───┬───┐  ┐
+                  │  │   │   │   │   │   │   │  │
+                  │  ├───┼───┼───┼───┼───┼───┤  │
+    f_2d.shape[0] │  │   │   │   │   │   │   │  │
+        (=3)      │  ├───┼───┼───┼───┼───┼───┤  │
+                  │  │   │   │   │   │   │   │  │
+                  └  └───┴───┴───┴───┴───┴───┘  ┘
+    ```
+
+
++ To declare a 3D scalar field, set the dimensions of the three axes:
+
+  ```python
+  f_3d = ti.field(ti.f32, shape=(32, 16, 8))  # 3D field
+  ```
+
+  You can think of this 3D scalar field as a 32&times;16 2D field whose elements are arrays of length 8:
+
+  ```
+                    f_3d.shape[1]
+                        (=16)
+                   ┌───────────────┐
+
+                ┌  ┌───┬╌╌╌╌╌╌╌┬───┐  ┐
+                │  │ * │       │   │  │
+                │  ├───┼╌╌╌╌╌╌╌┼───┤  │
+  f_3d.shape[0] │  ╎   ╎       ╎   ╎  │
+      (=32)     │  ├───┼╌╌╌╌╌╌╌┼───┤  │
+                │  │   │       │   │  │
+                └  └───┴╌╌╌╌╌╌╌┴───┘  ┘
+
+      ┌───┬───┬───┬───┬───┬───┬───┬───┐
+  * = │   │   │   │   │   │   │   │   │
+      └───┴───┴───┴───┴───┴───┴───┴───┘
+      └───────────────────────────────┘
+             f_1d.shape[2] = 8
+
+  ```
```

Comment:
I feel like this is too verbose. The readers should understand what a high-dimensional array is. Let's remove the 3D case here.